cpp/cmake/Modules/compute_matrix_product.py

```
@@ -0,0 +1,245 @@
+            else:
+                if isinstance(last, str):
+                    match = IDENTIFIER_RE.search(last)
+                    assert match
+                    underscores = match.group("underscores")
+                    rest = match.group("rest")
+                    path_repr = "".join(
+                        f"[{json.dumps(i)}]" for i in path[:-1]
+                    )
+
+                    if warn_used and used and underscores:
+                        warnings.warn(
```

Ruff (0.15.12)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)